### ISSUE TITLE
[video_player] Half of the video is green

### ISSUE THREAD

**Issue by mthcalixto** [flutter/flutter#77021]
I'm using the **video_player** plugin and I'm trying to run this video on it but in my app half is on **green screen**.

What can that be?

How it looks on my device: (Android 6.0)

<details><summary>Information:</summary><p>

**Code:**
```
Container(
                height: widget.size.height,
                width: widget.size.width,
                color: primary,
                child: Stack(
                  children: <Widget>[
                    ClipRect(
                        child: OverflowBox(
                            maxWidth: double.infinity,
                            maxHeight: double.infinity,
                            alignment: Alignment.center,
                            child: FittedBox(
                                fit: BoxFit.cover,
                                alignment: Alignment.center,
                                child: Container(
                                    width: widget.size.width,
                                    height: widget.size.height,
                                    child: AspectRatio(
                                        aspectRatio: 16.0 / 9.0,
                                        child: VideoPlayer(
                                            _videoPlayerController))))))
                  ],
                ),
              )
```

**Image Screen:**
![](https://i.imgur.com/nSDF1Rf.jpg)

**AndroidManifest.xml:**
`android:hardwareAccelerated="true"`

**Size video:**
`1024x576`

**Flutter doctor -v:**
```
flutter doctor -v
[√] Flutter (Channel dev, 1.26.0-12.0.pre, on Microsoft Windows [versÃ£o 10.0.18363.1139], locale
    pt-BR)
    • Flutter version 1.26.0-12.0.pre at G:\src\flutter
    • Framework revision a706cd2112 (7 weeks ago), 2021-01-14 18:20:26 -0500
    • Engine revision effb529ece
    • Dart version 2.12.0 (build 2.12.0-224.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\Matheus\AppData\Local\Android\Sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:\Users\Matheus\AppData\Local\Android\Sdk
    • ANDROID_SDK_ROOT = C:\Users\Matheus\AppData\Local\Android\Sdk
    • Java binary at: G:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Android Studio (version 4.1.0)
    • Android Studio at G:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.53.2)
    • VS Code at C:\Users\Matheus\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.19.0

[√] Connected device (1 available)
    • Lenovo K10a40 (mobile) • 0026760053 • android-arm • Android 6.0 (API 23)

• No issues found!
```

</p></details>

**Comment by darshankawar** [comment]
@matheusqro 
Can you upgrade to latest dev and try again to see if issue persists or not ?
If yes, provide updated `flutter doctor -v` and a [complete minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that shows the issue.

**Comment by matheusqro** [comment]
> @matheusqro
> Can you upgrade to latest dev and try again to see if issue persists or not ?
> If yes, provide updated `flutter doctor -v` and a [complete minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that shows the issue.

I updated:
```
flutter doctor -v
[√] Flutter (Channel dev, 1.27.0-8.0.pre, on Microsoft Windows [versÃ£o 10.0.18363.1139], locale
    pt-BR)
    • Flutter version 1.27.0-8.0.pre at G:\src\flutter
    • Framework revision b7d4806243 (11 days ago), 2021-02-19 09:22:45 -0800
    • Engine revision 6993cb229b
    • Dart version 2.13.0 (build 2.13.0-30.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\Matheus\AppData\Local\Android\Sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:\Users\Matheus\AppData\Local\Android\Sdk
    • ANDROID_SDK_ROOT = C:\Users\Matheus\AppData\Local\Android\Sdk
    • Java binary at: G:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at G:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at G:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.53.2)
    • VS Code at C:\Users\Matheus\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.20.0

[√] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 88.0.4324.190
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 88.0.705.81

• No issues found!
```
But the same thing happens, the background is still green.

This plugin does not happen that:
**https://pub.dev/packages/fijkplayer**
But it leaves the FPS of the video horrible on the device.

**Comment by darshankawar** [comment]
@matheusqro Thanks for code sample and video.
Although I don't have the said physical device, I tried your code sample on emulator (api 23) on which I see below:

<img width="347" alt="Screenshot 2021-03-03 at 11 53 36 AM" src="https://user-images.githubusercontent.com/67046386/109762798-34641b80-7c17-11eb-9d30-583e71fc224b.png">

Can you also switch to latest master and try again to see if issue persists or not ?
It seems to be device specific.

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.27.0-9.0.pre.175, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 1.27.0-9.0.pre.175 at
      /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 97295dc9a8 (66 minutes ago), 2021-03-03 00:19:03 -0500
    • Engine revision cd0e54348b
    • Dart version 2.13.0 (build 2.13.0-93.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    •
      Android 6.0 (API 23) (emulator)
    • macOS (desktop)                    • macos         • darwin-x64     • Mac
      OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)                       • chrome        • web-javascript •
      Google Chrome 88.0.4324.192

! Doctor found issues in 1 category.

```

</details>

**Comment by matheusqro** [comment]
Friend do you know what may be causing this error?

I updated it and it still doesn't work.

<details><summary>Flutter doctor -v</summary><p>
[√] Flutter (Channel master, 2.1.0-11.0.pre.89, on Microsoft Windows [versÃ£o 10.0.18363.1139],
    locale pt-BR)
    • Flutter version 2.1.0-11.0.pre.89 at G:\src\flutter
    • Framework revision ddcb8d7d6d (3 hours ago), 2021-03-03 14:11:30 -0800
    • Engine revision cd0e54348b
    • Dart version 2.13.0 (build 2.13.0-93.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\Matheus\AppData\Local\Android\Sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:\Users\Matheus\AppData\Local\Android\Sdk
    • ANDROID_SDK_ROOT = C:\Users\Matheus\AppData\Local\Android\Sdk
    • Java binary at: G:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at G:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at G:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.53.2)
    • VS Code at C:\Users\Matheus\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.20.0

[√] Connected device (3 available)
    • Lenovo K10a40 (mobile) • 0026760053 • android-arm    • Android 6.0 (API 23)
    • Chrome (web)           • chrome     • web-javascript • Google Chrome 88.0.4324.190
    • Edge (web)             • edge       • web-javascript • Microsoft Edge 88.0.705.81

• No issues found!
</p></details>

I believe it is the **plugin**  that is causing conflict with my device and not the flutter.

I believe it is the **ExoPlayer**.

Topics with the same error: #55090 #46460 #46628 #34642

**Comment by darshankawar** [comment]
@matheusqro 
I don't know about the ijkplayer plugin since it seems to be 3rd party. 
I am going ahead and keeping this open since I can't replicate it due to lack of specific device.

**Comment by darshankawar** [comment]
May be related / similar to https://github.com/flutter/flutter/issues/34642, https://github.com/flutter/flutter/issues/55090, https://github.com/flutter/flutter/issues/46460, https://github.com/flutter/flutter/issues/46628

**Comment by matheusqro** [comment]
@darshankawar Should I use another plugin then for the specific device?

Instead of using "video_player" do I use the other plugin for the device?

**Comment by ds84182** [comment]
This should be fixed by https://github.com/flutter/engine/pull/24888. Some devices decode video into large surfaces. Flutter Engine was not handling the transformation matrix correctly previously, but it is handled correctly now.

**Comment by darshankawar** [comment]
> This should be fixed by [flutter/engine#24888](https://github.com/flutter/engine/pull/24888). Some devices decode video into large surfaces. Flutter Engine was not handling the transformation matrix correctly previously, but it is handled correctly now.

@mthcalixto Can you check with latest stable version and using the same said device (Android 6.0) to see if you still see the reported behavior or not ?
